Assets/Scripts/DayNightCycle/DayAndNightControl.cs

```diff
@@ -0,0 +1,257 @@
+            frames++;
+            //MC loads settings
+            if (PhotonNetwork.isMasterClient)
+            {
+                GameSettings.Time.currentTime = currentTime;
+                GameSettings.Time.dayLength = DayLength;
+                GameSettings.Time.pause = pause;
+                Debug.Log("Current Master Client time: " + GameSettings.Time.currentTime);
+            }
+
+        }
+        
+        void UpdateLight()
+        {
+            Quaternion tilt = Quaternion.AngleAxis(sunTilt, Vector3.forward);
+            Quaternion rot = Quaternion.AngleAxis((CurrentTime01 * 360) - 90, Vector3.right);
+
+            directionalLight.transform.rotation = tilt * rot; // Yes axial tilt
+            directionalLight.transform.Rotate(Vector3.up, sunRotationOffset - 90, Space.World);
+            moon.transform.forward = -directionalLight.transform.forward;
+
+            if (timecycle)
+            {
+                // Sun & moon's color and brightness
+                if (timecycle.overrideSunlight)
+                {
+                    directionalLight.color = timecycle.sunlightColor.Evaluate(CurrentTime01);
+                    directionalLight.intensity = timecycle.sunlightColor.Evaluate(CurrentTime01).a * timecycle.maxSunlightIntensity;
+                }
+                if (timecycle.overrideMoonlight)
+                {
+                    Light moonLight = moon.GetComponent<Light>();
+                    moonLight.color = timecycle.moonlightColor.Evaluate(CurrentTime01);
+                    moonLight.intensity = timecycle.moonlightColor.Evaluate(CurrentTime01).a * timecycle.maxMoonlightIntensity;
+                }
+
+                // Environment lighting
+                if (timecycle.overrideEnvironmentLighting)
+                {
+                    switch (timecycle.lightingOverrideMode)
+                    {
+                        case TimecycleProfile.AmbientLightingOverrideMode.Gradient:
+                            RenderSettings.ambientSkyColor = timecycle.skyColor.Evaluate(CurrentTime01);
+                            RenderSettings.ambientEquatorColor = timecycle.equatorColor.Evaluate(CurrentTime01);
+                            RenderSettings.ambientGroundColor = timecycle.groundColor.Evaluate(CurrentTime01);
+                            break;
+                        case TimecycleProfile.AmbientLightingOverrideMode.Color:
+                            RenderSettings.ambientLight = timecycle.lightingColor.Evaluate(CurrentTime01);
+                            break;
+                    }
+                }
+
+                // Fog
+                if (timecycle.overrideFog)
+                {
+                    RenderSettings.fogColor = timecycle.fogColor.Evaluate(CurrentTime01);
+                    RenderSettings.fogDensity = timecycle.fogColor.Evaluate(CurrentTime01).a * timecycle.maxFogDensity;
+                }
+            }
+        }
+
+        public TimeOfDay GetTimeOfDay()
```

Comment:
@WattanaGaming

Reply:
@FransStreicher wdym? I only modified it to use enum

Reply:
@WattanaGaming https://github.com/AoTTG-2/AoTTG-2/commit/97b2e408a1d91d1659ca7014c6ebff0192d2efcd here you changed it to enum and added the 'unknown' line of code, I'm wondering whether we need it? I can add a = sign to the > signs in the if statements so that time will always be one of the other values(morning,afternoon,evening,midnight), in that case, I don't know if we will need unknown as after adding equals signs in the if statements, any currentTime will always have a value 0<=x<=1 so perhaps no need for an unknown returning.

Reply:
Seems I was overlooking that it's required, will just move the code of line to the top as Lithrun mentioned defaults are usually at the top

Reply:
I think the line of code is needed and makes sense to be where it is, will just update some of the if conditions.